apps/sports-helsinki/src/domain/search/venueSearch/hooks/useSearchPage.tsx

```diff
@@ -0,0 +1,125 @@
+import {
+  useSearchTranslation,
+  useIsSmallScreen,
+  useUnifiedSearch,
+  getLargeEventCardId,
+} from 'events-helsinki-components';
+import type { UnifiedSearchVenue } from 'events-helsinki-components';
+import { useRouter } from 'next/router';
+import React from 'react';
+import { useConfig } from 'react-helsinki-headless-cms';
+import { scroller } from 'react-scroll';
+import { toast } from 'react-toastify';
+import AppConfig from 'domain/app/AppConfig';
+import type { SearchPage } from 'domain/search/combinedSearch/types';
+import useUnifiedSearchListQuery from 'domain/unifiedSearch/useUnifiedSearchListQuery';
+import { removeQueryParamsFromRouter } from 'utils/routerUtils';
+import { SEARCH_ROUTES } from '../../../../constants';
+
+const BLOCK_SIZE = 10;
+
+function useSearchPage(): SearchPage {
```

Comment:
This is a superlong hook, I would perhaps divide it to several smaller ones if there is clearly logical parts.

Reply:
Earlier it was inside the super super long component. I can try to split it even more, but the most important thing is that it implements the `SearchPage` interface.

Reply:
After re-examination, I don't think I can split it much. It's all related to the venuesData given by the UnifiedSearch-query-hook. And like I said, the most important thing is that it should return a SearchPage object. The SearchPage determines the consistent features between the venues and events searches, which is really important.

Reply:
That's ok